### PR TITLE
Update to MXNet version 0.12.1.

### PR DESCRIPTION
**Pull request by mjdenkowski**
[MXNet 0.12.1](https://github.com/apache/incubator-mxnet/releases/tag/0.12.1) has some pretty important bug fixes, including one for making multi-CPU decoding work properly.  Since it's a minor version update, there are no breaking changes.

#### Pull Request Checklist ##
- [x] Changes are complete (if posting work-in-progress code, prefix your pull request title with '[WIP]'
until you can check this box.
- [x] Unit tests pass (`pytest`)
- [x] System tests pass (`pytest test/system`)
- [x] Passed code style checking (`./pre-commit.sh` or manual run of pylint & mypy)
- [x] You have considered writing a test
- [x] Updated major/minor version in `sockeye/__init__.py`. Major version bump if this is a backwards incompatible change.
- [x] Updated CHANGELOG.md

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -18,7 +18,7 @@ Recent developments and changes are tracked in our [changelog](https://github.co
 
 Sockeye requires:
 - **Python3**
-- [MXNet-0.12.0](https://github.com/dmlc/mxnet/tree/v0.12.0)
+- [MXNet-0.12.1](https://github.com/dmlc/mxnet/tree/v0.12.0)
```
Comment (mjdenkowski) [inline README.md]: Minor versions appear to continue on the same branch, hence v0.12.0 here.

File: test/system/test_seq_copy_sys.py [hunk 2]
```diff
@@ -54,7 +54,7 @@
     ("Copy:transformer:lstm",
      "--encoder transformer --num-layers 2:1 --rnn-cell-type lstm --rnn-num-hidden 64 --num-embed 32"
      " --rnn-attention-type mhdot --rnn-attention-num-hidden 32 --batch-size 16 --rnn-attention-mhdot-heads 1"
-     " --loss cross-entropy --optimized-metric perplexity --max-updates 5000"
+     " --loss cross-entropy --optimized-metric perplexity --max-updates 6000"
```
Comment (mjdenkowski) [inline test/system/test_seq_copy_sys.py]: This test sometimes failed with very close scores.  Looks like it just needed a few more updates.

File: README.md (hunk 1 above)
Comment (fhieber) [reply README.md]: But these links refer to git tags, correct? And there seems to be a tag for 0.12.1:
https://github.com/apache/incubator-mxnet/tree/0.12.1

Comment (mjdenkowski) [reply README.md]: Good catch!  We were linking to the branch here but the tag is more correct.

Comment (fhieber) [reply README.md]: This links to the tag as far as I can tell :)

**Pull request merged**